lib/order_optimizer/sku.rb

```diff
@@ -4,18 +4,18 @@ class Sku
 
     def initialize(id, quantity:, price_per_sku: nil, price_per_unit: nil, min_quantity: nil)
       @id = id
-      @quantity = quantity
-      @min_quantity = min_quantity
-
-      @price_per_unit = BigDecimal(price_per_unit, 2) if price_per_unit
-      @price_per_sku  = BigDecimal(price_per_sku,  2) if price_per_sku
+      @quantity = BigDecimal(quantity, 2)
+      @min_quantity = BigDecimal(min_quantity, 2) if min_quantity
 
       unless price_per_unit || price_per_sku
         raise ':price_per_sku or :price_per_unit must be set'
       end
 
-      @price_per_unit ||= price_per_sku.to_f / quantity
-      @price_per_sku  ||= quantity * price_per_unit
+      @price_per_unit = BigDecimal(price_per_unit, 2) if price_per_unit
+      @price_per_unit ||= BigDecimal(price_per_sku, 2) / quantity
```

Comment:
It's got a bit lost in the diff, but this is the crucial line: `@price_per_unit` will always be a `BigDecimal` now.